src/writer.rs

```diff
@@ -84,11 +92,49 @@ pub struct FdtWriter {
 ///
 /// This represents an area of physical memory reserved by the firmware and unusable by the OS.
 /// For example, this could be used to preserve bootloader code or data used at runtime.
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd)]
 pub struct FdtReserveEntry {
-    /// Physical address of the beginning of the reserved region.
-    pub address: u64,
-    /// Size of the reserved region in bytes.
-    pub size: u64,
+    address: u64,
+    size: u64,
+}
+
+impl FdtReserveEntry {
+    /// Create a memory reservation for the FDT.
+    ///
+    /// # Arguments
+    ///
+    /// * address: Physical address of the beginning of the reserved region.
+    /// * size: Size of the reserved region in bytes.
+    pub fn new(address: u64, size: u64) -> Result<Self> {
+        if address.checked_add(size).is_none() {
```

Comment:
We should probably also consider `size == 0` to be invalid.

Reply:
Good point, will add that check as well.